katalog/thanos/thanos-components/thanos-compact/thanos-compact.yaml

```diff
@@ -0,0 +1,102 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: thanos-compact
+  name: thanos-compact
+spec:
+  ports:
+    - port: 10902
+      protocol: TCP
+      targetPort: http
+      name: http-query
+  selector:
+    app: thanos-compact
+  sessionAffinity: None
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: thanos-compact
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: thanos-compact
+spec:
+  strategy:
+    type: Recreate
+  replicas: 1
+  selector:
+    matchLabels:
+      app: thanos-compact
+  template:
+    metadata:
+      labels:
+        app: thanos-compact
+      name: thanos-compact
+    spec:
+      terminationGracePeriodSeconds: 1200
+      serviceAccountName: thanos-compact
+      containers:
+        - name: thanos-compact
+          image: thanos
+          args:
+            - compact
+            - --log.level=info
+            - --data-dir=/var/thanos/compact
+            - --wait
+            - --consistency-delay=30m
+            - --objstore.config-file=/etc/thanos/config.yaml
+          ports:
+            - name: http
+              containerPort: 10902
+          livenessProbe:
+            httpGet:
+              path: /-/healthy
+              port: http
+              scheme: HTTP
+            periodSeconds: 90
+            successThreshold: 1
+            failureThreshold: 5
+            timeoutSeconds: 10
+          readinessProbe:
+            httpGet:
+              path: /-/ready
+              port: http
+              scheme: HTTP
+            periodSeconds: 90
+            successThreshold: 1
+            failureThreshold: 5
+            timeoutSeconds: 10
+          volumeMounts:
+            - name: data
+              mountPath: /var/thanos/compact
+            - name: thanos-storage
+              mountPath: /etc/thanos
+          resources:
+            requests:
+              cpu: 0
```

Comment:
request: 0 is the same as "unlimited". Are you sure we want this behavior?
If true, consider removing the `cpu: 0`